### ISSUE TITLE
Moving opensim-moco into opensim-core

### ISSUE THREAD

**Issue by chrisdembia** [opensim-org/opensim-core#2559]
The opensim-moco contains a lot of code that belongs better in opensim-core. Here's a list:

- [x] ModelProcessor #2712
- [x] TableProcessor #2713
- [x] analyze() #2718
- [x] visualize() #2717
- [x] DeGrooteFregly2016Muscle #2714
- [x] ModelFactory #2715
- [x] Stopwatch
- [x] Catch2 testing framework #2716
- [x] format()
- [x] SmoothSphereHalfspaceForce
- [x] ActivationCoordinateActuator
- [x] requiring numpy for python bindings.
- [x] Combine Moco's doxygen with opensim-core's.
- [ ] What to do with `build_on_*` scripts?
- [x] Should opensim-models still be a submodule?
- [ ] Merge CHANGELOG and README?
- [x] Update `opensim-cmd` to support running .omoco files.

@nickbianco feel free to add to this list.

Proposal in Google Doc: https://docs.google.com/document/d/10SiVJGqMJxiXtiDdI4Ozpxf5HTKHaixX_ynymYuJwk4/edit?usp=sharing

**Comment by aymanhab** [comment]
Moving DeGroote muscle to opensim-core is high priority so that users don't have to juggle muscle models to use/visualize results from Moco in GUI. What is involved @chrisdembia @nickbianco ?

**Comment by jenhicks** [comment]
@chrisdembia Is this issue up-to-date? I think one key task we talked about was formulating a plan/rough timeline for integration. Let us know how we can help. I chatted with Scott a couple weeks ago and he agreed that the integration sounded like a good plan (in terms of how we spend development resources/time). @aymanhab

**Comment by chrisdembia** [comment]
I've started working on requiring NumPy for the Python bindings.

**Comment by chrisdembia** [comment]
I created a branch in this repo `moco` into which I've merged the `opensim-moco` repository. I roughly followed these steps to perform the merge: https://stackoverflow.com/questions/1425892/how-do-you-merge-two-git-repositories

The merge had the conflicts listed below; I accepted the `opensim-core` version in these conflicts. These conflicts still need to be resolved through subsequent PRs.

warning: Cannot merge binary files: .github/.deploy_myosin_sourceforge_rsa.enc (HEAD vs. moco/master)
dependencies/CMakeLists.txt
cmake/CMakeLists.txt
README.md
CONTRIBUTING.md
CMakeLists.txt
CHANGELOG.md
.travis.yml
.gitignore
.github/workflows/continuous_integration.yml
.github/PULL_REQUEST_TEMPLATE.md
.github/.deploy_myosin_sourceforge_rsa.enc
.gitattributes
.clang-format
